src/Events/Editor/Full_Site/Templates.php

```diff
@@ -77,14 +78,47 @@ public function add_events_archive( array $query_result, $query, string $templat
 	 * @return WP_Block_Template A reference to the template object for the query.
 	 */
 	public function get_template_events_archive() {
-		$template_content = file_get_contents(
-			Tribe__Events__Main::instance()->plugin_path . '/src/Events/Editor/Full_Site/Templates/archive-events.html'
-		);
 
 		$template                 = new WP_Block_Template();
```

Comment:
Could use some guidance here...

It seems to me, that in our `get_template_events_archive()` we should be looking for a saved template, if none exists create it, then return that. 

Part of my belief for this is because in WP Core, on the `get_block_template`, and quite a few other places, [it is looking for the wp_id](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L693) to hydrate the object.

Also, we (assuming this is correct definition) are defining this a `custom` template. Which with that definition it [bypasses some of the creation](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L353-L360) logic. Also, on the front end the default save operation is a `PUT` (for some reason?) which is not hitting the `create_item()` callback.

Thoughts on that?

One other question - are we intending this `Archive Events` to be a single template? Multiple can't exist right? That is part of the assumption I'm going on.

Reply:
You're spot on. Using the `wp_id` to hydrate objects is a common pattern in WP Core, and it makes sense for us to align with that. First checking for a saved template is a good idea and should mitigate against unintentionally creating duplicates or overlooking previously set configurations for that template. If none exists, creating it ensures that we have a consistent fallback.

Regarding your question about the Archive Events template, yes, it's designed to be a singular entity. The logic ensures that we don't have multiple instances of the same template, which could lead to inconsistency and confusion. By ensuring only one instance, we can maintain a uniform experience across the board.